dts/bindings/modem/gsm-ppp.yaml

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2021 G-Technologies Sdn. Bhd.
+# SPDX-License-Identifier: Apache-2.0
+
+description: GSM PPP modem
+
+compatible: "gsm,ppp"
+
+include: uart-device.yaml
+
+properties:
+  label:
+    required: true
```

Comment:
Really not sure about this. I think we should try to just use DEVICE_DT_GET and avoid the need for requiring a label property. One of the main purposes of adding APIs like DEVICE_DT_GET was to make labels optional.

Reply:
Do you mean removing this completely or just make it optional?

Reply:
Keeping it optional.

Reply:
Changing this from true to false will result in a compilation error, don't really know why, have to look into it later

```
-- Found devicetree overlay: boards/frdm_uart2_dts.overlay
devicetree error: C:/Users/yongc/zephyrproject/zephyr/dts/bindings\modem\gsm-ppp.yaml (in 'label'): 'required' from included file overwritten ('True' replaced with 'False')
CMake Error at C:/Users/yongc/zephyrproject/zephyr/cmake/dts.cmake:240 (message):
  gen_defines.py failed with return code: 1
Call Stack (most recent call first):
  C:/Users/yongc/zephyrproject/zephyr/cmake/app/boilerplate.cmake:554 (include)
  C:/Users/yongc/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  C:/Users/yongc/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:35 (include_boilerplate)
  CMakeLists.txt:5 (find_package)
```

Reply:
Just remove `label:` from here, so it will be inherited from `uart-device.yaml`.